test/e2e_node/log_path_test.go

```diff
@@ -17,131 +17,156 @@ limitations under the License.
 						// this pod is expected to exit successfully
 						RestartPolicy: v1.RestartPolicyNever,
 						Containers: []v1.Container{
 							{
 								Image:   busyboxImage,
-								Name:    logContName,
-								Command: []string{"sh", "-c", "echo " + logString},
+								Name:    logContainerName,
+								Command: []string{"sh", "-c", "echo " + log},
 							},
 						},
 					},
 				}
+			}
 
-				podClient.Create(logPod)
-				err := framework.WaitForPodSuccessInNamespace(f.ClientSet, logPodName, ns)
-				framework.ExpectNoError(err, "Failed waiting for pod: %s to enter success state", logPodName)
-
-				// get containerID from created Pod
-				createdLogPod, err := podClient.Get(logPodName, metav1.GetOptions{})
-				logConID := kubecontainer.ParseContainerID(createdLogPod.Status.ContainerStatuses[0].ContainerID)
-				framework.ExpectNoError(err, "Failed to get pod: %s", logPodName)
-
-				expectedlogFile := logDir + "/" + logPodName + "_" + ns + "_" + logContName + "-" + logConID.ID + ".log"
-
+			makeLogCheckPod := func(podName, log, expectedLogPath string) *v1.Pod {
 				hostPathType := new(v1.HostPathType)
 				*hostPathType = v1.HostPathType(string(v1.HostPathFileOrCreate))
 
-				checkPod := &v1.Pod{
+				return &v1.Pod{
 					ObjectMeta: metav1.ObjectMeta{
-						Name: checkPodName,
+						Name: podName,
 					},
 					Spec: v1.PodSpec{
 						// this pod is expected to exit successfully
 						RestartPolicy: v1.RestartPolicyNever,
 						Containers: []v1.Container{
 							{
 								Image: busyboxImage,
-								Name:  checkContName,
+								Name:  podName,
 								// If we find expected log file and contains right content, exit 0
 								// else, keep checking until test timeout
-								Command: []string{"sh", "-c", "while true; do if [ -e " + expectedlogFile + " ] && grep -q " + logString + " " + expectedlogFile + "; then exit 0; fi; sleep 1; done"},
+								Command: []string{"sh", "-c", "while true; do if [ -e " + expectedLogPath + " ] && grep -q " + log + " " + expectedLogPath + "; then exit 0; fi; sleep 1; done"},
 								VolumeMounts: []v1.VolumeMount{
 									{
-										Name: logDirVolumeName,
+										Name: "logdir",
 										// mount ContainerLogsDir to the same path in container
-										MountPath: expectedlogFile,
+										MountPath: expectedLogPath,
 										ReadOnly:  true,
 									},
 								},
 							},
 						},
 						Volumes: []v1.Volume{
 							{
-								Name: logDirVolumeName,
+								Name: "logdir",
 								VolumeSource: v1.VolumeSource{
 									HostPath: &v1.HostPathVolumeSource{
-										Path: expectedlogFile,
+										Path: expectedLogPath,
 										Type: hostPathType,
 									},
 								},
 							},
 						},
 					},
 				}
+			}
+
+			createAndWaitPod := func(pod *v1.Pod) error {
+				podClient.Create(pod)
+				return framework.WaitForPodSuccessInNamespace(f.ClientSet, pod.Name, f.Namespace.Name)
+			}
+
+			var logPodName string
+			BeforeEach(func() {
```

Comment:
why two `BeforeEach`

Reply:
I mainly learned from another e2e_test file.

https://github.com/kubernetes/kubernetes/blob/197fc67693c2391dcbc652fc185ba85b5ef82a8e/test/e2e_node/security_context_test.go#L40

https://github.com/kubernetes/kubernetes/blob/197fc67693c2391dcbc652fc185ba85b5ef82a8e/test/e2e_node/security_context_test.go#L226

I think, in the future, there may be other test case to be added to log_path_test, which may have some other different initializing operations than Context("printed log to stdout"). 

For example, if we want to test the file based logs, which may be considered in the future. 

But the first `BeforeEach`, which initialize variable podClient, is needed in every test case.

If my concern is not necessary , I can modify this into one `BeforeEach`

Reply:
You are using single spec in this file, so you can use only one `BeforeEach`, this two `BeforeEach` will get execute in this spec, so you can make it one.

Reply:
ok，I will change this